### DISCUSSION TITLE
[Question]: How to keep the blog pages without the region code in the URL but maintain it for the other pages?

### DISCUSSION THREAD

**Discussion by gazpachu** [medusajs/nextjs-starter-medusa#387] [Q&A]
This could be a common use case, as the effort to maintain a blog/magazine in different languages/regions could only be possible for huge multinational corporations.

Since all the pages in this starter repo are under the `[countryCode]` folder, if we move the `blog` pages outside that folder, then the links in the top nav get broken because the country code is no longer available.

How can we keep the blog pages like this (domain.ltd/blog/my-post-title) while the other medusa pages keep the country code: `/es/cart`, `fr/checkout`, etc

**Comment by VariableVic** [comment]
I did not test this, but I think you could achieve this by doing the following:

Create a separate `layout.tsx` file in the `/blog` folder, with an alternative `<Nav>` component where you use regular `<Link>`s instead of `<LocalizedClientLink>` (which require the country code). 

The middleware will take care of redirecting users to the proper country code if there is none present in the url, so you can safely link to `/search` from the header menu. It will redirect to `/[countryCode]/search`. Make sure you edit `middleware.ts` to ignore `/blog/`.

Hope this guides you into the right direction! Let me know how it works out.

**Comment by gazpachu** [comment]
Thanks for the heads up, @VariableVic. In the end, aside from the `Layout` file and modifying the `middleware`, changing the `LocalizedClientLink` did the trick:

```
<Link href={`${countryCode ? `/${countryCode}` : ''}${href}`} {...props}>
  {children}
</Link>
```
